### PR TITLE
Raven db 4902 - WIP - do not merge

### PR DESCRIPTION
**Pull request by fitzchak**

### CODE REVIEW

File: src/Raven.Client/Documents/Commands/BlittableJsonContent.cs [hunk 1]
```diff
@@ -9,33 +10,25 @@ namespace Raven.Client.Documents.Commands
 {
     public class BlittableJsonContent : HttpContent
     {
-        private readonly BlittableJsonReaderObject _document;
-        private readonly JsonOperationContext _context;
+        private readonly Action<Stream> _writer;
 
-        public BlittableJsonContent(BlittableJsonReaderObject document, JsonOperationContext context)
+        public BlittableJsonContent(Action<Stream> writer)
```
Comment (ayende) [inline src/Raven.Client/Documents/Commands/BlittableJsonContent.cs]: Why do you do that?

File: src/Raven.Client/Http/RequestExecuter.cs [hunk 2]
```diff
@@ -122,10 +122,25 @@ private void UpdateTopologyCallback(object _)
             }
         }
 
-        public async Task ExecuteAsync<TResult>(JsonOperationContext context, RavenCommand<TResult> command)
+        public void Execute<TResult>(RavenCommand<TResult> command)
         {
             var choosenNode = ChooseNodeForRequest(command);
-            await ExecuteAsync(choosenNode, context, command);
+            JsonOperationContext context;
+            using (_contextPoll.AllocateOperationContext(out context))
```
Comment (ayende) [inline src/Raven.Client/Http/RequestExecuter.cs:129]: Why aren't you calling `Task ExecuteAsync<TResult>(RavenCommand<TResult> command)` here?

File: src/Raven.Client/Document/InMemoryDocumentSessionOperations.cs [hunk 3]
```diff
@@ -76,12 +77,12 @@ public abstract class InMemoryDocumentSessionOperations : IDisposable
         protected readonly Dictionary<object, InMemoryDocumentSessionOperations.DocumentMetadata> entitiesAndMetadata =
             new Dictionary<object, DocumentMetadata>(ObjectReferenceEqualityComparer<object>.Default);
 
-        protected readonly Dictionary<string, JsonDocument> includedDocumentsByKey = new Dictionary<string, JsonDocument>(StringComparer.OrdinalIgnoreCase);
+        internal readonly Dictionary<string, BlittableJsonReaderObject> IncludedDocumentsByKey = new Dictionary<string, BlittableJsonReaderObject>(StringComparer.OrdinalIgnoreCase);
```
Comment (ayende) [inline src/Raven.Client/Document/InMemoryDocumentSessionOperations.cs:80]: You cannot do that, you don't have the context at this point.

File: src/Raven.Client/Http/RequestExecuter.cs (hunk 2 above)
Comment (fitzchak) [inline src/Raven.Client/Http/RequestExecuter.cs:129]: I changed that

**Pull request closed without merging**